opentelemetry-etw-logs/src/logs/exporter.rs

```diff
@@ -0,0 +1,461 @@
+    //         Self::register_events(provider, *keyword);
+    //     }
+    // }
+
+    fn get_severity_level(&self, severity: Severity) -> tld::Level {
+        match severity {
+            Severity::Debug
+            | Severity::Debug2
+            | Severity::Debug3
+            | Severity::Debug4
+            | Severity::Trace
+            | Severity::Trace2
+            | Severity::Trace3
+            | Severity::Trace4 => tld::Level::Verbose,
+
+            Severity::Info | Severity::Info2 | Severity::Info3 | Severity::Info4 => {
+                tld::Level::Informational
+            }
+
+            Severity::Error | Severity::Error2 | Severity::Error3 | Severity::Error4 => {
+                tld::Level::Error
+            }
+
+            Severity::Fatal | Severity::Fatal2 | Severity::Fatal3 | Severity::Fatal4 => {
+                tld::Level::Critical
+            }
+
+            Severity::Warn | Severity::Warn2 | Severity::Warn3 | Severity::Warn4 => {
+                tld::Level::Warning
+            }
+        }
+    }
+
+    #[allow(dead_code)]
+    fn enabled(&self, level: u8, keyword: u64) -> bool {
+        // TODO: Use internal enabled check. Meaning of enable differs from OpenTelemetry and ETW.
+        // OpenTelemetry wants to know if level+keyword combination is enabled for the Provider.
+        // ETW tells if level+keyword combination is being actively listened. Not all systems actively
+        // listens for ETW events, but they do it on samples.
+        // This may be fixed by applying the OpenTelemetry logic in the callback function.
+        self.provider.enabled(level.into(), keyword)
+    }
+
+    pub(crate) fn export_log_data(
+        &self,
+        log_data: &opentelemetry_sdk::export::logs::LogData,
+    ) -> opentelemetry_sdk::export::logs::ExportResult {
+        let level =
+            self.get_severity_level(log_data.record.severity_number.unwrap_or(Severity::Debug));
+
+        let keyword = match self
+            .exporter_config
+            .get_log_keyword_or_default(log_data.instrumentation.name.as_ref())
+        {
+            Some(keyword) => keyword,
+            _ => return Ok(()),
+        };
+
+        if !self.provider.enabled(level.as_int().into(), keyword) {
+            return Ok(());
+        };
+
+        let event_tags: u32 = 0; // TBD name and event_tag values
+        let field_tag: u32 = 0;
+        let mut event = tld::EventBuilder::new();
+
+        // reset
+        event.reset(&self.event_name, level, keyword, event_tags);
+
+        event.add_u16("__csver__", 0x0401u16, tld::OutType::Hex, field_tag);
+
+        self.populate_part_a(&mut event, log_data, field_tag);
+
+        let (event_id, event_name) = self.populate_part_c(&mut event, log_data, field_tag);
+
+        self.populate_part_b(&mut event, log_data, level, event_id, event_name);
+
+        // Write event to ETW
+        event.write(&self.provider, None, None);
+
+        Ok(())
```

Comment:
there should be a failure scenario, when the size of payload exceed 64kb. Is that handled by tld library? or we should do it here.